app/src/org/commcare/dalvik/odk/provider/InstanceProvider.java

```diff
@@ -365,10 +385,91 @@ public int update(Uri uri, ContentValues values, String where, String[] whereArg
                 throw new IllegalArgumentException("Unknown URI " + uri);
         }
 
+        // If we've changed a particular form instance's status, and not
+        // created a new entry (hence count > 0 check), we need to mirror the
+        // change in that form's record.  NOTE: this conditional is crucial,
+        // since updating a form record in turn calls this update function, so
+        // we need to break the infinite loop by only updating the form record
+        // when the the status changes.
+        if (values.containsKey(InstanceColumns.STATUS) && count > 0) {
+            try {
+                linkToSessionFormRecord(uri);
```

Comment:
This needs to happen for some updating saves, but crashes on others.
Example of crash is when you try to save twice in a row while editing a form.
Example of when it is needed, when you open up a form to edit, complete it: the form record also needs to be updated to say it is complete.